crates/red_knot_python_semantic/src/types.rs

```diff
@@ -271,25 +276,55 @@ fn module_type_symbols<'db>(db: &'db dyn Db) -> smallvec::SmallVec<[ast::name::N
         .collect()
 }
 
-pub(crate) fn global_symbol<'db>(
-    db: &'db dyn Db,
-    lookup: SymbolLookup,
-    file: File,
-    name: &str,
-) -> Symbol<'db> {
-    // Not defined explicitly in the global scope?
-    // All modules are instances of `types.ModuleType`;
-    // look it up there (with a few very special exceptions)
-    symbol(db, lookup, global_scope(db, file), name).or_fall_back_to(db, || {
-        if module_type_symbols(db)
-            .iter()
-            .any(|module_type_member| &**module_type_member == name)
-        {
-            KnownClass::ModuleType.to_instance(db).member(db, name)
+/// Return the symbol for a member of `types.ModuleType`.
+pub(crate) fn module_type_symbol<'db>(db: &'db dyn Db, name: &str) -> Symbol<'db> {
+    if module_type_symbols(db)
+        .iter()
+        .any(|module_type_member| &**module_type_member == name)
+    {
+        KnownClass::ModuleType.to_instance(db).member(db, name)
+    } else {
+        Symbol::Unbound
+    }
+}
+
+/// Infer the public type of a symbol (its type as seen from outside its scope).
+fn symbol<'db>(db: &'db dyn Db, scope: ScopeId<'db>, name: &str) -> Symbol<'db> {
+    symbol_impl(db, scope, name, RequiresExplicitReExport::No)
+}
+
+/// Infers the public type of a symbol as seen from the same file.
+///
+/// If it's not defined explicitly in the global scope, it will look it up in `types.ModuleType`
+/// with a few very special exceptions.
+///
+/// Use [`imported_symbol`] to perform the lookup as seen from outside the file (e.g. via imports).
+pub(crate) fn global_symbol<'db>(db: &'db dyn Db, file: File, name: &str) -> Symbol<'db> {
```

Comment:
What's the difference to `symbol`? Oh, is it that `symbol` infers the symbol as seen in that scope?

Reply:
Yes, `symbol` is for a specific scope while `global_symbol` is for the global scope specifically.

Reply:
I see. It might be worth extending the `symbol` documentation to mention that it looks up the symbol in a specific scope.

Reply:
Given that `symbol` takes a `ScopeId` and a symbol name, I'm not sure how else it could be interpreted? I guess it could say "Infer the public type of a symbol (its type as seen from outside its scope) by scope and symbol name."? But to me that seems redundant with what's already clear from the signature.

Reply:
I've added a "... in the given `scope`." to be more explicit.